packages/typescript/test/function-declaration.test.tsx

```diff
@@ -38,6 +38,26 @@ it("can be a default export", () => {
     `);
 });
 
+it("can be an async function", () => {
+  expect(toSourceText(<FunctionDeclaration async export name="foo" />)).toBe(d`
+    export async function foo(): Promise<void> {
```

Comment:
Only adding the explicit return type when the function is async is odd to me? Can we just leave it off and let the compiler infer like the non-async case?

Reply:
Yeah this is a good point